micrometer-meter-provider/src/main/java/io/opentelemetry/contrib/metrics/micrometer/ScheduledCallbackRegistrar.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.metrics.micrometer;
+
+import java.util.Objects;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * An implementation of {@link CallbackRegistrar} that uses a {@link ScheduledExecutorService} to
+ * execute the registered callbacks on the specified schedule.
+ */
+public final class ScheduledCallbackRegistrar implements CallbackRegistrar {
+  private final ScheduledExecutorService scheduledExecutorService;
+  private final long period;
+  private final TimeUnit timeUnit;
+
+  ScheduledCallbackRegistrar(
+      ScheduledExecutorService scheduledExecutorService, long period, TimeUnit timeUnit) {
+    this.scheduledExecutorService = scheduledExecutorService;
+    this.period = period;
+    this.timeUnit = timeUnit;
+  }
+
+  public static ScheduledCallbackRegistrarBuilder builder(
+      ScheduledExecutorService scheduledExecutorService) {
+    Objects.requireNonNull(scheduledExecutorService, "scheduledExecutorService");
+    return new ScheduledCallbackRegistrarBuilder(scheduledExecutorService);
+  }
+
+  @Override
+  public CallbackRegistration registerCallback(Runnable callback) {
+    if (callback != null) {
+      ScheduledFuture<?> future =
+          scheduledExecutorService.scheduleAtFixedRate(callback, period, period, timeUnit);
+      return () -> future.cancel(false);
```

Comment:
I don't know if this is an issue, but if it's done this way then all the callbacks will probably be called at slightly different times; meaning some when metrics collection happens some meters might have a somewhat up-to-date value, while some others might be a little bit stale. Is that okay?